scripts/force_layer_download.py

```diff
@@ -0,0 +1,235 @@
+        if not isinstance(body, list):
+            raise ClientException("expecting list")
+        return [t["timeline_id"] for t in body]
+
+    async def timeline_spawn_download_remote_layers(self, tenant_id, timeline_id, ongoing_ok=False):
+
+        resp = await self.sess.post(
+            f"/v1/tenant/{tenant_id}/timeline/{timeline_id}/download_remote_layers",
+        )
+        body = await resp.json()
+        if resp.status == 409:
+            if not ongoing_ok:
+                raise ClientException("download already ongoing")
+            pass  # response body has same shape for ongoing and newly created
+        elif not resp.ok:
+            raise ClientException(f"{resp}")
+
+        if not isinstance(body, dict):
+            raise ClientException("expecting dict")
+
+        return body
+
+    async def timeline_poll_download_remote_layers_status(
+        self,
+        tenant_id,
+        timeline_id,
+    ):
+        resp = await self.sess.get(
+            f"/v1/tenant/{tenant_id}/timeline/{timeline_id}/download_remote_layers",
+        )
+        body = await resp.json()
+
+        if not resp.ok:
+            raise ClientException(f"{resp}")
+
+        return body
+
+    async def timeline_download_remote_layers(
+        self,
+        tenant_id,
+        timeline_id,
+    ):
+        spawn_res = await self.timeline_spawn_download_remote_layers(
+            tenant_id, timeline_id, ongoing_ok=False
+        )
+        while True:
+            st = await self.timeline_poll_download_remote_layers_status(tenant_id, timeline_id)
+            logging.info("{tenant_id}:{timeline_id} state is: {st}")
+
+            if spawn_res["task_id"] != st["task_id"]:
+                raise ClientException("download task ids changed while polling")
+
+            if st["state"] != "Completed":
+                await asyncio.sleep(10)
+                continue
+
+            return st
+
+
+class Completed(dict[str, Any]):
+    pass
+
+
+async def do_timeline(client: Client, tenant_id, timeline_id):
+    completed = await client.timeline_download_remote_layers(tenant_id, timeline_id)
+    assert completed["state"] == "Completed"
+    return Completed(completed)
+
+
+async def main(args):
+    client = Client(args.pageserver_http_endpoint)
+
+    with open(args.report_output, "w") as report_out:
+        exit_code = await main_impl(args, report_out, client)
+
+    await client.close()
+
+    return exit_code
+
+
+async def main_impl(args, report_out, client: Client):
+    """
+    Returns OS exit status.
+    """
+
+    tenant_and_timline_ids: List[Tuple[str, str]] = []
+    # fill  tenant_and_timline_ids based on spec
+    for spec in args.what:
+        comps = spec.split(":")
+        if comps == ["ALL"]:
+            logging.info("get tenant list")
+            tenant_ids = await client.get_tenant_ids()
+            tasks = [
+                asyncio.create_task(client.get_timeline_ids(tenant_id)) for tenant_id in tenant_ids
```

Comment:
NIT: you don't need to `create_task`. They will be created inside the `asyncio.gather` in any case.